src/parseWithPointers.ts

```diff
@@ -1,30 +1,34 @@
-import { DiagnosticSeverity, IDiagnostic } from '@stoplight/types';
+import { DiagnosticSeverity, IDiagnostic, Optional } from '@stoplight/types';
 import {
   determineScalarType,
-  Kind,
   load as loadAST,
   parseYamlBoolean,
   parseYamlFloat,
   parseYamlInteger,
-  ScalarType,
-  YAMLAnchorReference,
   YAMLException,
-  YamlMap,
-  YAMLNode,
-  YAMLScalar,
-  YAMLSequence,
 } from '@stoplight/yaml-ast-parser';
 import { buildJsonPath } from './buildJsonPath';
 import { SpecialMappingKeys } from './consts';
 import { lineForPosition } from './lineForPosition';
-import { IParseOptions, YamlParserResult } from './types';
+import {
+  IParseOptions,
+  Kind,
+  ScalarType,
+  YAMLAnchorReference,
+  YAMLMap,
+  YAMLNode,
+  YamlParserResult,
+  YAMLScalar,
+  YAMLSequence,
+} from './types';
+import { isObject } from './utils';
 
 export const parseWithPointers = <T>(value: string, options?: IParseOptions): YamlParserResult<T | undefined> => {
   const lineMap = computeLineMap(value);
   const ast = loadAST(value, {
     ...options,
     ignoreDuplicateKeys: true,
-  });
+  }) as YAMLNode;
```

Comment:
Why is this cast now required?

Reply:
Cause `loadAST` returns type from `yaml-st-parser`